### ISSUE TITLE
Search for tags using regex

### ISSUE THREAD

**Issue by piranna** [WyriHaximus/github-action-get-previous-tag#46]
Allow to filter tags using a regex as https://github.com/marketplace/actions/find-latest-tag does, instead of just only using a prefix.

**Comment by WyriHaximus** [comment]
Good one, will put this and a bunch of other enhancements into 1.4

**Comment by WyriHaximus** [comment]
Just tagged [`v2.0.0`](https://github.com/WyriHaximus/github-action-get-previous-tag/releases/tag/v2.0.0) and it includes the `pattern` option, which takes a glob pattern. Would that meet your requirements?

**Comment by RemiBardon** [comment]
What is the pattern syntax? The README gives `v*[0-9].*[0-9].*[0-9]` as an example but I’ve never seen such syntax (or it’s very wrong if it’s a regular expression).

Edit 1: Looking at the code, it seems to be passed as-is to `git for-each-ref --sort=-refname --sort=-creatordate --count 1 --format="%(refname:short)" "refs/tags/${tagPattern}"`. I’d be surprised if git used a non-standard syntax.

Edit 2: `git for-each-ref --help` says: `If one or more <pattern> parameters are given, only refs are shown that match against at least one pattern, either using fnmatch(3) or literally, in the latter case matching completely or from the beginning up to a slash.`. This means `git for-each-ref` uses `fnmatch` under the hood, and `*[0-9].*` is interpreted as “any sequence of characters, followed by one digit, followed by a literal dot, followed by any sequence of characters”.

---

In conclusion, `pattern` indeed uses a standard globbing syntax (as mentioned by the README), but the confusing `v*[0-9].*[0-9].*[0-9]` should be replaced by `v[0-9]*.[0-9]*.[0-9]*` (same result, but more correct and robust).

**Comment by WyriHaximus** [comment]
> In conclusion, `pattern` indeed uses a standard globbing syntax (as mentioned by the README), but the confusing `v*[0-9].*[0-9].*[0-9]` should be replaced by `v[0-9]*.[0-9]*.[0-9]*` (same result, but more correct and robust).

If you want, you can PR that and I'll be happy to merge that in. Thinking about that, maybe it's a good idea to link to documentation about how the pattern is used and what can be supplied from git's POV.

**Comment by RemiBardon** [comment]
> If you want, you can PR that and I'll be happy to merge that in. Thinking about that, maybe it's a good idea to link to documentation about how the pattern is used and what can be supplied from git's POV.

I didn't have time for it and will likely not in the following days but I'll keep that in mind if no one else does it before me.

**Comment by WyriHaximus** [comment]
> > If you want, you can PR that and I'll be happy to merge that in. Thinking about that, maybe it's a good idea to link to documentation about how the pattern is used and what can be supplied from git's POV.
> 
> I didn't have time for it and will likely not in the following days but I'll keep that in mind if no one else does it before me.

Works for me, currently addressing #81 and ensuring everything is up to date again and green before fixing that.
